NetFabric.Hyperlinq/Conversion/AsValueEnumerable/AsValueEnumerable.Enumerable.cs

```diff
@@ -88,6 +94,7 @@ internal Enumerator(IEnumerable<TSource> enumerable)
                 public readonly void Reset() 
                     => enumerator.Reset();
 
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Hi, @aalmada, I'm just curious - why is inlining back in here after it was removed in #231? Have you observed degradation in performance?

Reply:
Hi @mphelt, I see you're paying attention. :) 
Lately, I've been trying to figure out how _StructLinq_ can perform better in some operations. One of the reasons was that, just like System.Linq, I was disposing inside `MoveNext`, when it returns false. This was slowing down some operations, not because of the call but, because of the required checks in `MoveNext`.
Because `Dispose` is only called once, I thought that having it inlined would make `MoveNext` more complex and so I removed the attribute. 
I kept thinking about it and then realized that I could remove the `Dispose` call from `MoveNext`. Every LINQ query has to be materialized using a `foreach` or an aggregation operator. These will call Dispose immediately after MoveNext returns `false` so, there's no need to call Dispose twice. 
In this particular case, I added back the attribute because it only calls the `Dispose`of the source enumerator. I'm not sure if I should added back at some other places.
I should have documented this in the PR but you're the first one to comment. I'll try to do a better job next time.

Reply:
Thanks for the brief explanation.
I try to pay attention to all PRs in your repo :)
In this particular case I wasn't sure why have you done that only in this 1 place - thus I asked.

By the way - have you considered updating some file or files with benchmark results with every PR that you make?

Reply:
Actually, I was planning to start committing the benchmarks results. 😊
I have [another repository](https://github.com/NetFabric/LinqBenchmarks) where I've been posting results for benchmarks comparing LINQ to other implementations. 
Now that I know how to do it, I'm going to copy the results generation code from the other repository.
I can't promise to update these on every PR because they take all night to run but I'll try to update these often.